pyvista/plotting/axes_assembly.py

```diff
@@ -465,43 +478,212 @@ def label_color(self, color: ColorLike):  # numpydoc ignore=GL08
+        return self.get_part_prop('color')[_AxisEnum.y :: 3]
 
     @y_color.setter
     def y_color(self, color: ColorLike | Sequence[ColorLike]):  # numpydoc ignore=GL08
-        self._set_axis_color(_AxisEnum.y, color)
+        self.set_part_prop('color', color, axis=_AxisEnum.y.value)
 
     @property
     def z_color(self) -> tuple[Color, Color]:  # numpydoc ignore=RT01
         """Color of the z-axis shaft and tip."""
-        return self._get_axis_color(_AxisEnum.z)
+        return self.get_part_prop('color')[_AxisEnum.z.value :: 3]
 
     @z_color.setter
     def z_color(self, color: ColorLike | Sequence[ColorLike]):  # numpydoc ignore=GL08
-        self._set_axis_color(_AxisEnum.z, color)
+        self.set_part_prop('color', color, axis=_AxisEnum.z.value)
+
+    def set_part_prop(
+        self,
+        name: str,
+        value: float | str | ColorLike | Sequence[float | str | ColorLike],
+        axis: Literal['x', 'y', 'z', 0, 1, 2, 'all'] = 'all',
+        part: Literal['shaft', 'tip', 0, 1, 'all'] = 'all',
+    ):
+        """Set :class:`~pyvista.Property` attributes for the axes shafts and/or tips.
+
+        This is a generalized setter method which sets the value of a specific
+        :class:`~pyvista.Property` attribute for any combination of axis shaft or tip
+        parts.
+
+        Parameters
+        ----------
+        name : str
+            Name of the :class:`~pyvista.Property` attribute to set.
+
+        value : float | str | ColorLike | Sequence[float | str | ColorLike]
+            Value to set the attribute to. If a single value, set all specified axes
+            shaft(s) or tip(s) :class:`~pyvista.Property` attributes to this value.
+            If a sequence of values, set the specified parts to these values.
+
+        axis : str | int, default: 'all'
+            Set :class:`~pyvista.Property` attributes for a specific part of the axes.
+            Specify one of:
+
+            - ``'x'`` or ``0``: only set the property for the x-axis.
+            - ``'y'`` or ``1``: only set the property for the y-axis.
+            - ``'z'`` or ``2``: only set the property for the z-axis.
+            - ``'all'``: set the property for all three axes.
+
+        part : str | int, default: 'all'
+            Set the property for a specific part of the axes. Specify one of:
+
+            - ``'shaft'`` or ``0``: only set the property for the axes shafts.
+            - ``'tip'`` or ``1``: only set the property for the axes tips.
+            - ``'all'``: set the property for axes shafts and tips.
+
+        Examples
+        --------
+        Set :attr:`~pyvista.Property.ambient` for all axes shafts and tips to a
+        single value.
+
+        >>> import pyvista as pv
+        >>> axes_actor = pv.AxesAssembly()
+        >>> axes_actor.set_part_prop('ambient', 0.7)
+        >>> axes_actor.get_part_prop('ambient')
+        _AxesPropTuple(x_shaft=0.7, y_shaft=0.7, z_shaft=0.7, x_tip=0.7, y_tip=0.7, z_tip=0.7)
```

Comment:
I am not sure I like the NamedTuple here. On the one hand, the named values are helpful since it's not obvious that the order should be `shaft-xyz` then `tip-xyz`. The order could be especially confusing since the symmetric `labels` param has the axes "grouped"  together as `(+x. -x, +y, -y, +z, -z)`, so both of these 6-tuples use different ordering conventions.

But, the named tuple  also makes it hard to read the values and compare them, since the text gets in the way and is cluttering things a bit.

Should we get rid of it?

Reply:
As a reviewer, I didn't find it difficult to read. `NamedTuple` is a good way to organize information, so I'm in favor of using it. +1 keep it as it is. If users complain about any inconveniences they experience when using the site, let's reconsider it.